### ISSUE TITLE
react/prop-types does not validate missing propTypes for functions in version 7.17.0

### ISSUE THREAD

**Issue by johnarnfield** [yannickcr/eslint-plugin-react#2526]
Seems like this is still an issue in version 7.17.0

Missing prop types do not raise errors in a function component with a return of:

```
return (
    ordering.length > 0 && (
      <Paginator items={ordering} pageSize={10} />
    )
)
```

But do with a function component with a return of:

```
  if (ordering.length > 0) {
    return false;
  }

  return (
    <Paginator items={ordering} pageSize={10} />
  )
```

and:

```
  return ordering.length > 0 ? (
    <Paginator items={ordering} pageSize={10} />
  ) : null;
```
_Originally posted by @johnarnfield in https://github.com/yannickcr/eslint-plugin-react/issues/2134#issuecomment-566035325_
